lib/internal/blob.js

```diff
@@ -0,0 +1,201 @@
+'use strict';
+
+const {
+  ArrayFrom,
+  SymbolIterator,
+  ObjectSetPrototypeOf,
+  Symbol,
+  Uint8Array,
+} = primordials;
+
+const { createBlob } = internalBinding('buffer');
+
+const {
+  JSTransferable,
+  kClone,
+  kDeserialize,
+} = require('internal/worker/js_transferable');
+
+const {
+  isAnyArrayBuffer,
+  isArrayBufferView,
+} = require('internal/util/types');
+
+const {
+  customInspectSymbol: kInspect,
+} = require('internal/util');
+const { inspect } = require('internal/util/inspect');
+
+const {
+  codes: {
+    ERR_INVALID_ARG_TYPE,
+    ERR_BUFFER_TOO_LARGE,
+    ERR_OUT_OF_RANGE,
+  }
+} = require('internal/errors');
+
+const {
+  validateObject,
+  validateString,
+  validateUint32,
+  isUint32,
+} = require('internal/validators');
+
+const kHandle = Symbol('kHandle');
+const kType = Symbol('kType');
+const kLength = Symbol('kLength');
+
+let Buffer;
+
+function lazyBuffer() {
+  if (Buffer === undefined)
+    Buffer = require('buffer').Buffer;
+  return Buffer;
+}
+
+function isBlob(object) {
+  return object?.[kHandle] !== undefined;
+}
+
+function getSource(source, encoding) {
+  if (isBlob(source))
+    return [source.size, source[kHandle]];
+
+  if (typeof source === 'string') {
+    source = lazyBuffer().from(source, encoding);
+  } else if (isAnyArrayBuffer(source)) {
+    source = new Uint8Array(source);
+  } else if (!isArrayBufferView(source)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'source',
+      [
+        'string',
+        'ArrayBuffer',
+        'SharedArrayBuffer',
+        'Buffer',
+        'TypedArray',
+        'DataView'
+      ],
+      source);
+  }
+
+  // We copy into a new Uint8Array because the underlying
+  // BackingStores are going to be detached and owned by
+  // the Blob. We also don't want to have to worry about
+  // byte offsets.
+  source = new Uint8Array(source);
+  return [source.byteLength, source];
+}
+
+class InternalBlob extends JSTransferable {
+  constructor(handle, length, type = '') {
+    super();
+    this[kHandle] = handle;
+    this[kType] = type;
+    this[kLength] = length;
+  }
+}
+
+class Blob extends JSTransferable {
+  constructor(sources = null, options) {
+    if (sources === null ||
+        typeof sources[SymbolIterator] !== 'function' ||
+        typeof sources === 'string') {
+      throw new ERR_INVALID_ARG_TYPE('sources', 'Iterable', sources);
+    }
+    if (options !== undefined)
+      validateObject(options, 'options');
+    const {
+      encoding = 'utf8',
+      type = ''
+    } = { ...options };
+
+    let length = 0;
+    const sources_ = ArrayFrom(sources, (source) => {
+      const { 0: len, 1: src } = getSource(source, encoding);
+      length += len;
+      return src;
+    });
+
+    // This is a MIME media type but we're not actively checking the syntax.
+    // But, to be fair, neither does Chrome.
+    validateString(type, 'options.type');
```

Comment:
This should set type to `''` if it has characters out of range `/[^\u{0020}-\u{007E}]/u` and lowercase.

```suggestion
    validateString(type, 'options.type');
    if (/[^\u{0020}-\u{007E}]/u.test(type)) type ='';
    type = StringPrototypeToLowerCase(type);
```

Reply:
Maybe add a `else` to avoid lower casing the empty string.
```suggestion
    validateString(type, 'options.type');
    if (RegExpPrototypeTest(/[^\u{0020}-\u{007E}]/u, type)) type ='';
    else type = StringPrototypeToLowerCase(type);
```